nilearn/_utils/testing.py

```diff
@@ -24,11 +28,11 @@
 from . import logger
 
 try:
-    from nose.tools import assert_raises_regexp
+    from nose.tools import assert_raises_regex
```

Comment:
slight niggle: assert_raises_regex does not exist for python 2.7 because it is called assert_raises_regexp (with a 'p' at the end) we should do something like this:

```python
try:
    from nose.tools import assert_raises_regex
except ImportError
    try:
        from nose.tools import assert_raises_regexp as assert_raises_regex
    except ImportError:
        # for Python 2.6
```

Side-comment: very annoyingly PEP8 is not stable with respect to search and replace which means that most of the assert_raises_regexp -> assert_raises_regex replacements in this PR have introduced PEP8 violations...